packages/angular/ssr/node/src/app-engine.ts

```diff
@@ -43,30 +43,37 @@ export class AngularNodeAppEngine {
   }
 
   /**
-   * Retrieves HTTP headers for a request associated with statically generated (SSG) pages,
-   * based on the URL pathname.
+   * Serves a prerendered page for a given request.
    *
-   * @param request - The incoming request object.
-   * @returns A `Map` containing the HTTP headers as key-value pairs.
-   * @note This function should be used exclusively for retrieving headers of SSG pages.
-   * @example
-   * ```typescript
-   * const angularAppEngine = new AngularNodeAppEngine();
+   * This method examines the incoming request to determine if it corresponds to a known prerendered page.
+   * If a matching page is found, it returns a `Response` object containing the page content; otherwise, it returns `null`.
    *
-   * app.use(express.static('dist/browser', {
-   *   setHeaders: (res, path) => {
-   *     // Retrieve headers for the current request
-   *     const headers = angularAppEngine.getPrerenderHeaders(res.req);
+   * @param request - The incoming HTTP request for a prerendered page.
+   * @returns A promise that resolves to a `Response` object containing the prerendered page content, or `null`
+   * if no matching page is found.
+   */
+  serve(request: IncomingMessage): Promise<Response | null> {
+    return this.angularAppEngine.serve(createWebRequestFromNodeRequest(request));
+  }
+
+  /**
+   * Handles incoming HTTP requests by serving prerendered content or rendering the page.
    *
-   *     // Apply the retrieved headers to the response
-   *     for (const [key, value] of headers) {
-   *       res.setHeader(key, value);
-   *     }
-   *   }
-     }));
-  * ```
-  */
-  getPrerenderHeaders(request: IncomingMessage): ReadonlyMap<string, string> {
-    return this.angularAppEngine.getPrerenderHeaders(createWebRequestFromNodeRequest(request));
+   * This method first attempts to serve a prerendered page. If the prerendered page is not available,
+   * it falls back to rendering the requested page using server-side rendering. The function returns
+   * a promise that resolves to the appropriate HTTP response.
+   *
+   * @param request - The incoming HTTP request for a prerendered page.
+   * @param requestContext - Optional additional context for rendering, such as request metadata.
+   * @returns A promise that resolves to the HTTP response object resulting from the request handling,
+   * or null if no matching content is found.
+   */
+  async process(request: IncomingMessage, requestContext?: unknown): Promise<Response | null> {
+    const webRequest = createWebRequestFromNodeRequest(request);
+
+    return (
+      (await this.angularAppEngine.serve(webRequest)) ??
+      (await this.angularAppEngine.render(webRequest, requestContext))
+    );
```

Comment:
Consider: Is there a reason not to delegate to `this.angularAppEngine.process`? Would that reduce duplication in the logic?